### PR TITLE
Add worker vCPU info to fsadmin report capacity command

### PR DESCRIPTION
**Pull request by Xenorith**
provide more information about worker state

### CODE REVIEW

File: core/common/src/main/java/alluxio/wire/WorkerInfo.java [hunk 1]
```diff
@@ -130,6 +131,14 @@ public String getRevision() {
     return mRevision;
   }
 
+  /**
+   * @return the git revision at the time of building the worker
+   */
+  @ApiModelProperty(value = "Number of available processors on the worker")
```
Comment (jiacheliu3) [inline core/common/src/main/java/alluxio/wire/WorkerInfo.java:137]: what does this do?

File: core/server/worker/src/main/java/alluxio/worker/block/BlockMasterClient.java [hunk 2]
```diff
@@ -310,7 +310,9 @@ public void register(final long workerId, final List<String> storageTierAliases,
 
     final RegisterWorkerPOptions options =
         RegisterWorkerPOptions.newBuilder().addAllConfigs(configList)
-            .setBuildVersion(buildVersion).build();
+            .setBuildVersion(buildVersion)
+            .setNumVCpu(Runtime.getRuntime().availableProcessors())
```
Comment (jiacheliu3) [inline core/server/worker/src/main/java/alluxio/worker/block/BlockMasterClient.java:314]: `Runtime.getRuntime().availableProcessors()` does this work in a container?
From https://github.com/Alluxio/alluxio/issues/17021 seems in some cases the issue still exists.

File: shell/src/test/java/alluxio/cli/fsadmin/report/CapacityCommandTest.java [hunk 3]
```diff
@@ -59,6 +59,7 @@ public void longCapacity() throws IOException {
       String output = new String(outputStream.toByteArray(), StandardCharsets.UTF_8);
       // CHECKSTYLE.OFF: LineLengthExceed - Much more readable
       List<String> expectedOutput = Arrays.asList("Capacity information for all workers: ",
+          "    Total vCPUs: 0",
```
Comment (jiacheliu3) [inline shell/src/test/java/alluxio/cli/fsadmin/report/CapacityCommandTest.java:62]: Update the cpu count to a non-zero number in `prepareLongInfoList()` and assert here?

File: core/common/src/test/java/alluxio/wire/WorkerInfoTest.java [hunk 4]
```diff
@@ -92,6 +92,7 @@ public static WorkerInfo createRandom() {
     String version = String.format("%d.%d.%d", random.nextInt(10),
         random.nextInt(20), random.nextInt(10));
     String revision = DigestUtils.sha1Hex(RandomStringUtils.random(10));
+    int numVCpu = random.nextInt(128);
```
Comment (jiacheliu3) [inline core/common/src/test/java/alluxio/wire/WorkerInfoTest.java:95]: hmmm why is a random cpu number here pass all UTs? Like there is one `checkEquality()` you didn't update right?

File: shell/src/test/java/alluxio/cli/fsadmin/report/CapacityCommandTest.java (hunk 3 above)
Comment (jiacheliu3) [reply shell/src/test/java/alluxio/cli/fsadmin/report/CapacityCommandTest.java:62]: there might be other UTs you can add, like in `BlockMasterTest` check a registered worker will have CPU recorded and summed at the BlockMaster. I'll leave this up to your judgement, depending on the importance of this feature and how much we rely on real system tests instead.

**Review by jiacheliu3 — APPROVED** [review]
Mostly LGTM with comments mostly as suggestions. So I'll just stamp ahead of time and trust the author with the rest. Thanks for the work!

File: core/server/worker/src/main/java/alluxio/worker/block/BlockMasterClient.java (hunk 2 above)
Comment (Xenorith) [reply core/server/worker/src/main/java/alluxio/worker/block/BlockMasterClient.java:314]: yup i am well aware of this issue

**Comment by Xenorith** [comment]
alluxio-bot, merge this please

**Comment by Xenorith** [comment]
alluxio-bot, cherry-pick this to branch-2.10 please

**Pull request merged**